nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/adapter/InstallerThread.java

```diff
@@ -79,120 +87,127 @@ public void run() {
+        if (domain.getSystemApplicationReferencedFrom(serverEnvironment.getInstanceName(),
+                ServerEnvironmentImpl.DEFAULT_ADMIN_CONSOLE_APP_NAME) != null) {
             // Application is already installed
             adapter.setStateMsg(AdapterState.APPLICATION_INSTALLED_BUT_NOT_LOADED);
             return;
         }
 
         // Set the adapter state
         adapter.setStateMsg(AdapterState.INSTALLING);
-        if (log.isLoggable(Level.FINE)) {
-            log.log(Level.FINE, "Installing the Admin Console Application...");
-        }
 
-        //create the application entry in domain.xml
-        ConfigCode code = new ConfigCode() {
-            @Override
-            public Object run(ConfigBeanProxy... proxies) throws PropertyVetoException, TransactionFailure {
-                SystemApplications sa = (SystemApplications) proxies[0];
-                Application app = sa.createChild(Application.class);
-                sa.getModules().add(app);
-                app.setName(AdminConsoleAdapter.ADMIN_APP_NAME);
-                app.setEnabled(Boolean.TRUE.toString());
-                app.setObjectType("system-admin"); //TODO
-                app.setDirectoryDeployed("true");
-                app.setContextRoot(contextRoot);
-                try {
-                    app.setLocation("${com.sun.aas.installRootURI}/lib/install/applications/" + AdminConsoleAdapter.ADMIN_APP_NAME);
-                } catch (Exception me) {
-                    // can't do anything
-                    throw new RuntimeException(me);
-                }
-                Module singleModule = app.createChild(Module.class);
-                app.getModule().add(singleModule);
-                singleModule.setName(app.getName());
-                Engine webe = singleModule.createChild(Engine.class);
-                webe.setSniffer("web");
-                Engine weldEngine = singleModule.createChild(Engine.class);
-                weldEngine.setSniffer("weld");
-                Engine sece = singleModule.createChild(Engine.class);
-                sece.setSniffer("security");
-                singleModule.getEngines().add(webe);
-                singleModule.getEngines().add(weldEngine);
-                singleModule.getEngines().add(sece);
-                Server s = (Server) proxies[1];
-                List<ApplicationRef> arefs = s.getApplicationRef();
-                ApplicationRef aref = s.createChild(ApplicationRef.class);
-                aref.setRef(app.getName());
-                aref.setEnabled(Boolean.TRUE.toString());
-                aref.setVirtualServers(getVirtualServerList()); //TODO
-                arefs.add(aref);
-                return true;
+        logger.log(Level.FINE, "Installing the Admin Console Application...");
+
+        // Create the application entry in domain.xml
+        ConfigCode code = proxies -> {
```

Comment:
Quite big anonymous class, maybe it should have a name and own file.

Reply:
Thanks for advice. Please DO NOT MERGE. I'll change it later.

Reply:
Done.